gtsam/slam/dataset.h

```diff
@@ -260,6 +277,14 @@ struct SfmData {
   SfmTrack track(size_t idx) const {
     return tracks[idx];
   }
+
+  void add_track(SfmTrack t) {
```

Comment:
Please change the argument to a const reference, unless there's a good reason not to.